### PR TITLE
Make navbar icons links

### LINKED ISSUE #159: Make navbar icons actual links
- home should go to root_path

- the icon buttons should all go to box_requests_path, but with varying filters already applied

- status icon should go to box_requests_path
- review icon should go to box_requests_path for box_requests that don't have any associated boxes (needs a scope on box_request model) and reviewed_by_id is EITHER current_user.id or nil
- design icon should go to box_requests_path for box_requests where designed_by_id is EITHER current_user.id or nil, so could be those assigned to "me" or those that need to be claimed (needs a scope on box_request model)
- package icon should go to box_requests_path for box_requests where packaged_by_id is EITHER current_user.id or nil(needs a scope on box_request model)
- shipping icon should go to box_requests_path for box_requests where shipped_by_id  is EITHER current_user.id or nil (needs a scope on box_request model)

Nav bar only- VOC

### PR DESCRIPTION
**Pull request by sarslanoglu**
Resolves #159 

### Description

Navbar icons where pointing only status. Right now all of them is using filters and current_user_id if possible

### Type of change

Links of navbar items are changed
Additional check is added to box_request controller for user_id existence

### How Has This Been Tested?

Tested manually

### CODE REVIEW

File: app/models/box_request.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@ class BoxRequest < ApplicationRecord
   delegate :name, to: :reviewed_by, prefix: :reviewer, allow_nil: true
 
   scope :requested, ->(){ where(reviewed_by_id: nil) }
-  scope :designed, ->(){ joins(:box).where("boxes.aasm_state = ?", "desgined") }
+  scope :designed, ->(){ joins(:box).where("boxes.aasm_state = ?", "designed") }
```
Comment (h-m-m) [inline app/models/box_request.rb:21]: Nice typo catch!

**Review by maebeale — COMMENTED** [review]
@sarslanoglu thank you!!! 

this task gets at some core usability, so thank you for digging in to where the links should point. 

i don't think we want to filter all of these on current_user bc then an admin wouldn't be able to see the work of everyone. 

ideally people with the respective permission will see either unclaimed items or items they've claimed but not completed, unless they're a superadmin and can see everything. (i think the closest to this we're getting currently is if they have Permission::MANAGE_USERS).

do you have suggestions for how to move forward? (e.g. should we approve of this PR and add a new issue, or do you feel comfortable updating what's here to get a little closer?)

**Comment by oconnelc** [comment]
# Problem Description
I'm not sure if anyone else has seen this, but I pulled this branch into my work space and it was working  great, until I completely restarted all of the processes (the docker-compose, the rails server and web pack). I found that I was unable to login. The code correctly identified the state as not being logged in, but when I tried to go the root page (http://localhost:3000), I got the following error:

## Screenshot
image

# Analysis
It looks like the current user isn't getting set. It's possible that my local work space wasn't in a good state and the problem has nothing to do with this pull request. 

# Workaround
When I went back to the commit, I was able to get the login screen and login. Once I was logged in, I checked out my branch that had merged this branch in and I was able to use the application with this fix.

**Comment by sarslanoglu** [comment]
@oconnelc It was my mistake to not support not logged in users at first place. It should be fixed by second [commit](https://github.com/rubyforgood/voices-of-consent/pull/414/commits/1d2f9e307a221a9b9c6c714c29307c8bb2fd3069)

@maebeale I changed it to support both not signed in users also add support for admin users. Basically admins can open with current_user_id to see everything

**Comment by h-m-m** [comment]
@sarslanoglu Do you think we should add a test for each of the non-logged in, logged in without admin, and logged in as admin behavior? I'm thinking a test to make sure the request works end-to-end would be enough, as that would work as a regression test for what @oconnelc found

**Comment by maebeale** [comment]
Totally, @h-m-m !

**Comment by sarslanoglu** [comment]
@maebeale @h-m-m [Last commit](https://github.com/rubyforgood/voices-of-consent/pull/414/commits/55e6381c288f9946952886c657287bd55af333c3) is dealing with navbar tests

File: app/controllers/box_requests_controller.rb [hunk 2]
```diff
@@ -9,6 +9,7 @@ def index
     if params[:filter_by].present?
       (filter_attr = params[:filter_by])
       @box_requests = @box_requests.public_send(filter_attr)
+      @box_requests.where(reviewed_by_id: params[:current_user_id]) if params[:current_user_id].present?
```
Comment (maebeale) [inline app/controllers/box_requests_controller.rb:12]: @sarslanoglu should this be `@box_requests = @box_requests.where...` ?

**Review by maebeale — CHANGES_REQUESTED** [review]
@sarslanoglu thank you for this PR! 
i merged in some other navbar PR's, which created a merge issue.

**Pull request merged**